tests/modules/restful_test/src/Plugin/resource/entity_test/main/v1/Main__1_8.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful_test\Plugin\resource\entity_test\main\v1\Main__1_8.
+ */
+
+namespace Drupal\restful_test\Plugin\resource\entity_test\main\v1;
+
+use Drupal\restful\Plugin\resource\DataInterpreter\DataInterpreterInterface;
+
+/**
+ * Class Main__1_8.
+ *
+ * @package Drupal\restful_test\Plugin\resource
+ *
+ * @Resource(
+ *   name = "main:1.8",
+ *   resource = "main",
+ *   label = "Main",
+ *   description = "Export the entity test 'main' bundle.",
+ *   authenticationOptional = TRUE,
+ *   dataProvider = {
+ *     "entityType": "entity_test",
+ *     "bundles": {
+ *       "main"
+ *     },
+ *   },
+ *   majorVersion = 1,
+ *   minorVersion = 8
+ * )
+ */
+class Main__1_8 extends Main__1_0 {
+
+  /**
+   * Overrides ResourceEntity::publicFields().
+   */
+  protected function publicFields() {
+    $public_fields = parent::publicFields();
+
+    $class = get_class($this);
+    $public_fields['random_rel'] = array(
```

Comment:
The thing is that for `$node->uid` you _could_ use this method. But you are going to be better with something already existing like:

```php
$public_fields['author'] = array(
  'property' => 'uid',
  'class' => '\Drupal\restful\Plugin\resource\Field\ResourceFieldEntityReference',
  'resource' => array(
    'name' => 'users',
    'majorVersion' => 1,
    'minorVersion' => 0,
  ),
);
```

That is very similar, but it assumes that the result of the field is an ID of an entity so it will try to load the entity if needed (and keep track of the entity type, …).